src/main/java/ai/shreds/application/ApplicationPurchaseService.java

```diff
@@ -0,0 +1,114 @@
+import org.springframework.messaging.support.MessageBuilder;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ApplicationPurchaseService implements ApplicationPurchaseServiceInputPort {
+
+    private static final Logger log = LoggerFactory.getLogger(ApplicationPurchaseService.class);
+    private final DomainSupplierValidationPort supplierValidationPort;
+    private final DomainProductValidationPort productValidationPort;
+    private final DomainPurchaseTransactionPort purchaseTransactionPort;
+    private final AdapterKafkaProducer kafkaProducer;
+    private final InfrastructureEntityMapper entityMapper;
+
+    @Override
+    public AdapterPurchaseResponseDTO processPurchaseTransaction(AdapterPurchaseRequestDTO request) {
+        try {
+            // Validate supplier
+            boolean isSupplierValid = supplierValidationPort.validateSupplier(request.getSupplierId());
+            if (!isSupplierValid) {
+                log.error("Invalid Supplier ID: {}", request.getSupplierId());
+                return new AdapterPurchaseResponseDTO(400, null, "Invalid Supplier ID");
+            }
+
+            // Validate products
+            List<String> productIds = request.getProducts().stream().map(AdapterProductDTO::getProductId).collect(Collectors.toList());
+            boolean areProductsValid = productValidationPort.validateProducts(productIds);
+            if (!areProductsValid) {
+                log.error("Invalid Product IDs: {}", productIds);
+                return new AdapterPurchaseResponseDTO(400, null, "Invalid Product IDs");
+            }
+
+            // Map to domain entity
+            DomainPurchaseTransactionEntity domainTransaction = entityMapper.mapAdapterToDomain(request);
+
+            // Save transaction
+            purchaseTransactionPort.savePurchaseTransaction(domainTransaction);
+
+            // Send Kafka message asynchronously
+            sendKafkaMessageAsync("ProductAdded", domainTransaction);
+
+            // Prepare response
+            AdapterPurchaseDataDTO responseData = entityMapper.mapDomainToAdapter(domainTransaction);
+            return new AdapterPurchaseResponseDTO(200, responseData, null);
+        } catch (DomainPurchaseTransactionException e) {
+            log.error("Error occurred during transaction processing: ", e);
+            return new AdapterPurchaseResponseDTO(500, null, "Error occurred during transaction processing");
+        } catch (Exception e) {
+            log.error("Internal Server Error: ", e);
+            return new AdapterPurchaseResponseDTO(500, null, "Internal Server Error");
+        }
+    }
+
+    private Future<SendResult<String, String>> sendKafkaMessageAsync(String topic, DomainPurchaseTransactionEntity transaction) {
+        String message = transaction.toString();
+        Message<String> msg = MessageBuilder.withPayload(message).build();
+        Future<SendResult<String, String>> future = kafkaProducer.sendMessage(topic, message);
```

Comment:
MAke sendMessage() in the AdapterKafkaProducer class return Future <org. springframework. kafka. support. SendResult<java. lang. String,java. lang. String>> instead of void